core/deployment/src/main/java/io/quarkus/deployment/builditem/StaticInitConfigSourceProviderBuildItem.java

```diff
@@ -2,10 +2,13 @@
 
 import io.quarkus.builder.item.MultiBuildItem;
 
-public final class AdditionalStaticInitConfigSourceProviderBuildItem extends MultiBuildItem {
+/**
+ * Provides a way to register a ConfigSourceProvider in STATIC INIT.
+ */
+public final class StaticInitConfigSourceProviderBuildItem extends MultiBuildItem {
```

Comment:
Do we know if this class is being used by anybody but us? If so, we can't just rename it, we'll need a new build item and deprecate the old one

Reply:
I don't believe so, but we can do it to be on the safe side.

Reply:
Yeah, let's be safe on this one as it has the potential to be pretty catastrophic for say Camel :)